arangod/Transaction/Manager.cpp

```diff
@@ -645,6 +697,14 @@ Result Manager::updateTransaction(TransactionId tid, transaction::Status status,
       // make sure everyone who asks gets the updated timestamp
       mtrx.updateExpiry();
       if (mtrx.finalStatus == status) {
+        if (tid.isFollowerTransactionId() &&
+            mtrx.finalStatus == transaction::Status::ABORTED && 
+            mtrx.intermediateCommits) {
+          // we are trying to abort a follower transaction (again) that had intermediate
+          // commits already. in this case we return a special error code, which makes
+          // the leader drop us as a follower for all shards in the transaction.
+          return res.reset(TRI_ERROR_CLUSTER_FOLLOWER_TRANSACTION_COMMIT_PERFORMED);
```

Comment:
the follower transaction is alreadx aborted though, not sure what this is trying to accomplish

Reply:
this check is too late and is only there when a second leader tries to abort